### DISCUSSION TITLE
What to use instead of Drv/SerialDriverPorts

### DISCUSSION THREAD

**Discussion by LincolnSand** [nasa/fprime#2600] [Q&A] [answered]
So, we have some code that uses an old version of fprime:
```
        @ output port for writing commands over UART to device
        output port serialWrite: Drv.SerialWrite

        @ receive serial data port to read data from UART
        async input port serialRecv: Drv.SerialRead

        @ serial buffer port for device to write into over UART
        output port serialBufferOut: Fw.BufferSend
```
But Drv/SerialDriverPorts no longer seems to exist?
Relevant error:
```
           output port serialWrite: Drv.SerialWrite
                                 ^
error: undefined symbol Drv
```
   I cannot find any update logs or issues that describe what should be used instead. Do you have any recommendations of what to use instead? Should we be using Drv/LinuxUartDriver now? I'm mostly just wondering how to migrate the above code to newer versions of fprime. Thank you for any help and I apologize if this question is stupid!

**Comment by thomas-bc** [comment]
I'm going to convert this issue into a [Discussion](https://github.com/nasa/fprime/discussions) as it seems to fit that category

**Comment by LeStarch** [comment, marked as answer]
The UART driver has been converted into a `Drv.ByteStreamDriver` thus it uses the `send` and `recv` ports.  See here: https://github.com/nasa/fprime/blob/devel/Drv/Interfaces/ByteStreamDriverInterface.fppi
